### PR TITLE
React

### PR DESCRIPTION
**Pull request by vamsisollem**
To-do with react

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  useEffect(() => {
+    fetchTasks();
+  }, []);
+
+  // fetch tasks every time when a new task is added 
+  useEffect(() => {
+    if (taskName) {
+      const newTask = createNewTask(taskName);
+
+      async function postData() {
```
Comment (banavasi) [inline src/App.jsx:42]: this can be a separate method. and call postData(taskName)

File: vite.config.js
Comment (banavasi) [inline vite.config.js:1]: Add path alias, where you can use @ or ~ to access src files

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect } from 'react';
+import Header from './components/header';
```
Comment (banavasi) [inline src/App.jsx:2]: use path alias, meaning access header components you need to use @/components/header

File: src/components/header.css
Comment (banavasi) [inline src/components/header.css:1]: All this css can be incorporated with tailwind. width: 300px; can be w-[300px]

File: src/components/header.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import './header.css'
```
Comment (banavasi) [inline src/components/header.jsx:2]: I think this is redundant

File: src/components/task.jsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+import './task.css';
```
Comment (banavasi) [inline src/components/task.jsx:2]: This seems redundant.

**Review by banavasi — CHANGES_REQUESTED** [review]

File: src/components/task.jsx (hunk 4 above)
Comment (vamsisollem) [reply src/components/task.jsx:2]: Do you want me to use header and task in the app itself ?

File: package.json
Comment (banavasi) [inline package.json:1]: why do you have so many dependencies?

File: src/components/task.jsx [hunk 5]
```diff
@@ -62,24 +62,25 @@ function Task({ tasks, fetchTasks }) {
   return (
     <div className='flex items-center flex-col'>
       {filteredTasks.map((task) => (
-        <div key={task.id} className='task-list'>
+        <div key={task.id} className='bg-very-dark-desaturated-blue text-very-light-gray mb-1 w-[500px] flex items-center p-2.5 rounded-md'>
           <input 
             type="checkbox"
```
Comment (banavasi) [inline src/components/task.jsx:67]: You can create a BaseInput component. Probably try using styled components

**Review by banavasi — APPROVED** [review]

**Pull request merged**